overlord/snapstate/handlers.go

```diff
@@ -607,8 +608,20 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	}
 	// only set userID if unset in snapst and if we actually have
 	// an associated user
```

Comment:
this comment probably needs updating

also: is it right to leave snapst.UserID alone if the user no longer exists but snapsup.UserID is unset?